### ISSUE TITLE
[ArrayDataSource]: "RangeError: Maximum call stack size exceeded" error when using with more than 100,000 items

### ISSUE THREAD

**Issue by vasilii-kovalev** [epam/UUI#2091]
## Steps to Reproduce

1. Go to this [Devbox](https://codesandbox.io/p/devbox/uui-array-data-source-call-stack-size-issue-vwv85d?file=%2Fsrc%2Fexample.tsx%3A8%2C1)
2. Observe the UI and console with a small amount of items (`ITEMS_COUNT` value is set to 10)
3. Change `ITEMS_COUNT` to 150,000
4. Observe the UI and console with a big amount of items

## Actual result

With a small amount of items, `PickerInput` is rendered correctly and there are no errors in the console.
With a big amount of items, a white screen appears and an error "RangeError: Maximum call stack size exceeded" occurs in the console

Stack trace:
```
chunk-33VLWUXS.js?v=7fec0a52:9426 Uncaught RangeError: Maximum call stack size exceeded
    at sortRec (chunk-33VLWUXS.js?v=7fec0a52:9426:19)
    at _Tree.sort (chunk-33VLWUXS.js?v=7fec0a52:9432:5)
    at ArrayListView.updateTree (chunk-33VLWUXS.js?v=7fec0a52:9631:43)
    at ArrayListView.update (chunk-33VLWUXS.js?v=7fec0a52:9601:12)
    at new ArrayListView (chunk-33VLWUXS.js?v=7fec0a52:9582:10)
    at chunk-33VLWUXS.js?v=7fec0a52:10218:50
    at mountMemo (chunk-S5TY363X.js?v=7fec0a52:12817:27)
    at Object.useMemo (chunk-S5TY363X.js?v=7fec0a52:13141:24)
    at useMemo (chunk-JQ3MWCGW.js?v=7fec0a52:1094:29)
    at ArrayDataSource.useView (chunk-33VLWUXS.js?v=7fec0a52:10218:43)
```

<img width="573" alt="image" src="https://github.com/epam/UUI/assets/10310491/abe29852-3bb9-48a2-bdeb-61a81dbc7d73">

## Expected result

In both cases, `PickerInput` is rendered correctly and there are no errors in the console.

## Environment

* UUI version: 5.7.0
* Browser name: Google Chrome v.123.0.6312.58
* Operating System: Windows 10 64-bit

## Notes

* If I understand correctly, @jakobz suspects, that the issue is related to a huge amount of parameters passed to `sortedItems.push` (see the screenshot in "Actual result")
* The error also occurs when using `AsyncDataSource`
* The error **does not** occur when using `LazyDataSource`, so it can be used as a workaround until the issue is fixed

**Comment by Kuznietsov** [comment]
Closes by https://github.com/epam/UUI/pull/1880

**Comment by NatalliaAlieva** [comment]
Released in 5.8.0 ver.
